src/components/nabla/common/NablaTokenPrice.tsx

```diff
@@ -5,14 +5,33 @@ export type TokenPriceProps = {
   address: string;
   prefix?: ReactNode;
   fallback?: ReactNode;
+  currentTokenAmount?: number;
+  formatByAmount?: boolean;
 };
 
-export function NablaTokenPrice({ address, prefix = null, fallback = null }: TokenPriceProps): JSX.Element | null {
+export function NablaTokenPrice({
+  currentTokenAmount,
+  address,
+  prefix = null,
+  fallback = null,
+  formatByAmount = false,
+}: TokenPriceProps): JSX.Element | null {
   const { data, isLoading } = useNablaTokenPrice(address);
   if (isLoading) return <NumberLoader />;
 
   if (data === undefined) return fallback;
 
+  if (formatByAmount) {
+    const currentAmountPrice = (Number(data.approximateStrings.atLeast2Decimals) * (currentTokenAmount || 0)).toFixed(
+      2,
+    );
```

Comment:
Let's generally avoid `number` for financial applications and only use the `Big.js` library or native `bigint`.

```suggestion
    const currentAmountPrice = (data.preciseBigDecimal.mul(currentTokenAmount ?? 0)).toFixed(
      2, 0
    );
```